src/client/render/secondstage.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+Minetest
+Copyright (C) 2010-2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+Copyright (C) 2017 numzero, Lobachevskiy Vitaliy <numzer0@yandex.ru>
+Copyright (C) 2020 appgurueu, Lars Mueller <appgurulars@gmx.de>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "secondstage.h"
+#include "client/client.h"
+#include "client/shader.h"
+#include "client/tile.h"
+
+PostProcessingStep::PostProcessingStep(video::E_MATERIAL_TYPE shader, const std::vector<u8> &_texture_map) :
+	texture_map(_texture_map)
+{
+	createMaterial(shader);
+}
+
+void PostProcessingStep::createMaterial(video::E_MATERIAL_TYPE shader)
+{
+	material.UseMipMaps = false;
+	material.ZBuffer = true;
+	material.ZWriteEnable = video::EZW_ON;
+	material.MaterialType = shader;
+	for (u32 k = 0; k < MYMIN(video::MATERIAL_MAX_TEXTURES, texture_map.size()); ++k) {
```

Comment:
don't we have a problem if `texture_map.size()` > `video::MATERIAL_MAX_TEXTURES`?
should this be a `sanity_check`?

Reply:
Changed to assert, because this count is managed compile time.